src/uct/ib/rc/accel/rc_mlx5_iface.c

```diff
@@ -266,6 +269,15 @@ static UCS_CLASS_INIT_FUNC(uct_rc_mlx5_iface_t, uct_md_h md, uct_worker_h worker
                              sizeof(struct mlx5_wqe_ctrl_seg)) /
                              sizeof(struct mlx5_wqe_data_seg));
 
+#if HAVE_IBV_EXP_DM
+    self->super.dm.am_short  = self->super.dm.dm ? uct_rc_mlx5_ep_am_short_dm :
```

Comment:
looks like you could update iface->ops directly. Then you would avoid extra call (thru pointer) when DM is enabled

Reply:
no, can't: EP's uses same ops instance for all EP instances, but we have to manage every EP based on available DM

Reply:
where do you manage it per EP? AFAIU you are using fallback to bcopy if no DM available